frappe/hooks.py

```diff
@@ -543,3 +543,7 @@
 	"Activity Log": 90,
 	"Route History": 90,
 }
+
+# Technically we have event handlers but `frappe` gets special treatment
+# SocketIO server uses this to identify which apps to lookup for event handlers.
+has_realtime_event_handlers = False
```

Comment:
why hooks configuration along with file path convention? if there is a file path convention, how expensive is it to crawl all apps and find realtime event handler files?

if we are going with hooks configuration, shouldn't it be a file path like other similar hooks?

Reply:
mainly to avoid hitting file system on every connection. _Most_ apps will not use this feature.

Reply:
do some caching magic? don't cache in developer mode?

Reply:
Makes sense, assuming it wont change in lifetime of socketio process. (fair assumption in prod)

Reply:
Added basic cache using object, no need to disable in dev mode IMO as `require` isn't gonna refetch code anyway (?) you'd need nodemon while developing socketio events handlers.